WorkflowUI/Tests/AdaptedEnvironmentScreenTests.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  AdaptedEnvironmentTests.swift
+//  BlueprintUI-Unit-Tests
+//
+//  Created by Kyle Van Essen on 6/28/21.
+//
+
+#if canImport(UIKit)
+
+import UIKit
+import XCTest
+@testable import WorkflowUI
+
+class AdaptedEnvironmentScreenTests: XCTestCase {
+    func test_wrapping() {
+        var environment: ViewEnvironment = .empty
+
+        let screen = TestScreen { environment = $0 }
```

Comment:
It will be, yeah. I think that's generally fine; since deeply nesting like this in practice won't really occur. This test is a bit of a degenerate test, designed to simulate many `adaptedEnvironments` that could be throughout your Screen tree.

> there are a few alternatives i can think of for accumulating the environment mutations

We do this in Blueprint, but its a _lot_ easier because the `AdaptedEnvironment` element there isn't generic around the element type; the generics make this much more difficult.